src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs

```diff
@@ -98,7 +98,8 @@ private void WriteIdProperty()
         {
             _writer.Line();
             _writer.WriteXmlDocumentationSummary($"Typed Resource Identifier for the container.");
-            _writer.Line($"public new {typeof(ResourceGroupResourceIdentifier)} Id => base.Id as {typeof(ResourceGroupResourceIdentifier)};");
+            var idType = _resourceContainer.OperationGroup.GetResourceIdentifierType(_resourceContainer.ResourceData, _context.Configuration.MgmtConfiguration);
```

Comment:
`_resourceContainer.ResourceData` is not used inside `GetResourceIdentifierType`, we may need further review of the `GetResourceIdentifierType` method.

Reply:
Removed the `mgmtObjectType`(`_resourceContainer.ResourceData`) parameter as it's not needed.